packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js

```diff
@@ -174,6 +188,27 @@ export function BlockSettingsDropdown( {
 	const parentBlockIsSelected =
 		selectedBlockClientIds?.includes( firstParentClientId );
 
+	// When a currentClientId is in use, treat the menu as a controlled component.
+	// This ensures that only one block settings menu is open at a time.
```

Comment:
My only comment at this point is: wouldn't be simpler to always use `DropdownMenu` in a controlled way? Something like

```
const open = currentClientId !== undefined && openedBlockSettingsMenu === currentClientId;
```

I believe that the logic would just be simpler, and the behaviour of the `DropdownMenu` component more predictable (furthermore, React usually **[warns](https://stackoverflow.com/questions/47012169/a-component-is-changing-an-uncontrolled-input-of-type-text-to-be-controlled-erro)** when this pattern is applied to `input` elements)

Reply:
It'd be simpler in this component, but due to current usage of the BlockSettingsDropdown, a `currentClientId` isn't always available as a `block` object isn't always provided. So we currently need to support both controlled and uncontrolled in this component.

In practice, at the moment, the List View provides a `block` object, however the [block settings menu in the toolbar doesn't](https://github.com/WordPress/gutenberg/blob/fec05ee8451ee1a8da43cdbf00485736e2b0a16a/packages/block-editor/src/components/block-toolbar/index.js#L152) (it provides `clientIds` for the selected blocks instead, which isn't quite the same as the concept of the `block` when used in the list view), so if we switch to always controlling the component, we wind up breaking the dropdown in the editor canvas:

https://github.com/WordPress/gutenberg/assets/14988353/680427c9-861d-4c41-9ecf-a779ebdd305b

Separately to this PR we could potentially look at refactoring other usage of the block settings dropdown to ensure a `block` object with `clientId` is always provided, and then we could have the component always operate in controlled mode.

For now, I'd prefer to try to contain the scope of this PR, so I think retaining this slightly complex ternary is probably a good step to go with for now, as it focuses on the main use case to solve which is with the behaviour in the list view.

Reply:
This is the block of code that I'm suggesting to add some more comments about it being a temp solution..